packages/lib-auth/src/views/oauth/components/shared/Popup/Popup.spec.js

```diff
@@ -0,0 +1,38 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+import sinon from 'sinon'
+
+import { Popup } from './Popup'
+
+const spy = sinon.spy()
+
+describe('Component > OAuthApp > Popup', function () {
+  it('should render without crashing', function () {
+    shallow(<Popup loginFunction={spy} />)
+  })
+
+  it('should render its children', function () {
+    const string = 'foobar'
+    const wrapper = shallow(
+      <Popup loginFunction={spy}>
+        <div>{string}</div>
+      </Popup>
+    )
+
+    expect(wrapper.text()).to.equal(string)
+  })
+
+  describe('login button', function () {
+    it('should exist', function () {
+      const wrapper = shallow(<Popup loginFunction={spy} />)
+      expect(wrapper.find('Button').length).to.equal(1)
+    })
+
+    it('should call the login prop when clicked', function () {
```

Comment:
This test is failing: `TypeError: Cannot read property 'token' of undefined`
in Popup.js on line 19:3.